src/reactviews/common/virtualizedList.ts

```diff
@@ -0,0 +1,188 @@
+            this._container.scrollTop + this._container.clientHeight
+        ) {
+            this._container.scrollTop =
+                itemBottom - this._container.clientHeight;
+        }
+    }
+
+    private updateFocusedItemIndex(newIndex: number) {
+        if (newIndex < 0 || newIndex >= this._items.length) {
+            return;
+        }
+        const oldItem = document.getElementById(
+            `${this._listId}-${this._focusedItemIndex}`,
+        );
+        if (oldItem) {
+            oldItem.setAttribute("tabindex", "-1");
+        }
+        const newItem = document.getElementById(`${this._listId}-${newIndex}`);
+        if (newItem) {
+            newItem.setAttribute("tabindex", "0");
+        }
+
+        this._focusedItemIndex = newIndex;
+        this.scrollToIndex(newIndex);
+        newItem?.focus();
+    }
+
+    private onScroll() {
+        const newOffset = Math.floor(
+            this._container.scrollTop / this._config.itemHeight,
+        );
+        if (newOffset !== this._scrollOffset) {
+            this._scrollOffset = newOffset;
+            this.renderList();
+        }
+    }
+
+    private renderList() {
+        const startIndex = Math.max(
+            this._scrollOffset - this._config.buffer,
+            0,
+        );
+        const endIndex = Math.min(
+            this._scrollOffset + this._visibleCount + this._config.buffer,
+            this._items.length,
+        );
+        // Remove existing visible items
+        Array.from(this._container.children)
+            .filter(
+                (child) =>
+                    child.tagName === "DIV" &&
+                    child !== this._container.firstChild,
+            )
+            .forEach((child) => this._container.removeChild(child));
+
+        for (let i = startIndex; i < endIndex; i++) {
+            const item = this._items[i];
+            const itemDiv = document.createElement("div");
+            itemDiv.id = `${this._listId}-${i}`;
+            this._renderItem(itemDiv, item);
+            itemDiv.style.position = "absolute";
+            itemDiv.style.height = `${this._config.itemHeight}px`;
+            itemDiv.style.width = "100%";
+            itemDiv.style.top = `${i * this._config.itemHeight}px`;
+            this._container.appendChild(itemDiv);
+            itemDiv.setAttribute(
+                "tabindex",
+                i === this._focusedItemIndex ? "0" : "-1",
+            );
+
+            // Set up click listener
+            this._eventManager.addEventListener(itemDiv, "click", (e) => {
+                this.updateFocusedItemIndex(i);
+                this._itemSelected(itemDiv, item);
+                e.preventDefault();
+                e.stopPropagation();
+            });
+
+            this._eventManager.addEventListener(itemDiv, "keydown", (e) => {
+                const event = e as KeyboardEvent;
+                let handled = false;
+                if (event.key === "ArrowDown") {
```

Comment:
extract all these keycode constants to a constants file, e.g. `Keys.ArrowDown` or `Keys.Space`

Reply:
Also, looks like the copyKeybind plugin uses the deprecated `.keyCode === 67` for "c"; can you update that?

Reply:
> Also, looks like the copyKeybind plugin uses the deprecated `.keyCode === 67` for "c"; can you update that?

Sure, I'll fix this one in a separate PR